great_expectations/expectations/metrics/column_map_metrics/column_values_between.py

```diff
@@ -73,6 +75,30 @@ def _pandas(
         if min_value is not None and max_value is not None and min_value > max_value:
             raise ValueError("min_value cannot be greater than max_value")
 
+        # Use a vectorized approach for native numpy dtypes
+        if column.dtype in [int, float] and allow_cross_type_comparisons is not True:
+            return cls._pandas_vectorized(
+                temp_column, min_value, max_value, strict_min, strict_max
+            )
+        elif (
+            column.dtype == np.dtype("datetime64[ns]")
+            and allow_cross_type_comparisons is not True
+        ):
+            if min_value is not None and not isinstance(max_value, datetime.datetime):
```

Comment:
Definitely a good catch on the `min_value` vs `max_value`.

Re: `parse_strings_as_datetimes`: this is another thorn, but I think there are two things going on here.

*Even if* `parse_strings_as_datetimes` is False (and now it should *always* be False as we remove that functionality), today, if you pass in a datetime as the *parameter* value (min_value), it will be stringified before this block of code to prepare for a json-compatible expectation configuration.

I think we know the *correct* fix is to make the expectation configuration parameters type aware.

Since they're not, what's actually happening here is that we're saying that if the *column* is a datetime, then we want the thing we're comparing it to to be a datetime as well, so we parse the parameter (`min_value` or `max_value`) *back* into a datetime.

Reply:
I see. I believe this conversion code is basically identical to the above. Can we put it in it's own function and call it in both?
It might be useful to add a comment that this may have been stringified and needs to be converted back. Maybe it's not necessary if that is common throughout the code but for me, who is relatively unfamiliar with this part of the codebase, it isn't clear to me.  Typing the function definition would help with this but maybe that would increase the scope of the PR too much.

Reply:
I propose we just add the comment at this point, since the `parse` logic above would be both slightly different (because of the try/catch) and not expected to survive a future stronger-typed parameter.